src/TagsInput/TagsInput.js

```diff
@@ -0,0 +1,151 @@
+// @flow
+
+import * as React from 'react';
+import { View, Platform } from 'react-native';
+import type { ViewLayoutEvent } from 'react-native/Libraries/Components/View/ViewPropTypes';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+
+import DeleteButton from './components/DeleteButton';
+import TagsContainer from './components/TagsContainer';
+import InputField from './components/InputField';
+import { StyleSheet } from '../PlatformStyleSheet';
+import { Text } from '../Text';
+
+import { MIN_WIDTH } from './constants';
+
+type Props = {|
+  +label: string,
+  +onChangeText: (value: string) => void,
+  +placeholder: string,
+  +selected?: string[],
+  +disabled?: boolean,
+  +fontSize: number,
+  +onClearPress?: () => void,
+  +value?: string,
+|};
+
+type State = {
+  value: string,
+  containerWidth: ?number,
+  inputWidth: ?number,
+};
+
+class TagInput extends React.Component<Props, State> {
+  inputRef: any; // @TODO find a way how to type this properly, the issue with mixing react and react-native
```

Comment:
I think we are using react > v16.3, so that could be helpful: https://reactjs.org/docs/refs-and-the-dom.html#creating-refs

Reply:
This was the first approach what I tried but, it didn't work.